src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -52,17 +64,42 @@ public DrsService(SnapshotService snapshotService,
                       DrsIdService drsIdService,
                       IamService samService,
                       ResourceService resourceService,
+                      ConfigurationService configurationService,
+                      KubeService kubeService,
                       PerformanceLogger performanceLogger) {
         this.snapshotService = snapshotService;
         this.fileService = fileService;
         this.drsIdService = drsIdService;
         this.samService = samService;
         this.resourceService = resourceService;
+        this.configurationService = configurationService;
+        this.kubeService = kubeService;
         this.performanceLogger = performanceLogger;
     }
 
-    public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand) {
+    public void decrement() {
+        int newValue = currentDRSRequests.decrementAndGet(); // how do I make sure this never becomes negative?
+        currentDRSRequests.set(newValue);
+    }
+
+    public void increment() {
+        int newValue = currentDRSRequests.incrementAndGet();
+        currentDRSRequests.set(newValue);
+        checkLookupMax();
+    }
+
+    public void checkLookupMax() throws TooManyRequestsException {
+        // make sure not too many requests are being made at once
+        int podCount = kubeService.getActivePodCount();
+        int maxDRSLookups = configurationService.getParameterValue(ConfigEnum.DRS_LOOKUP_MAX);
```

Comment:
Not sure it's required but you could unit test this code by making `ConfigEnum.DRS_LOOKUP_MAX` be `0`.